nemo/collections/tts/data/audio_trimming.py

```diff
@@ -0,0 +1,292 @@
+
+        self.pad_seconds = pad_seconds
+        self.volume_norm = volume_norm
+
+    def _detect_speech(self, audio: np.array) -> np.array:
+        # Center-pad the audio
+        audio = np.pad(audio, [self.frame_length // 2, self.frame_length // 2])
+
+        # [num_frames, frame_length]
+        audio_frames = librosa.util.frame(
+            audio, frame_length=self.frame_length, hop_length=self.frame_step
+        ).transpose()
+
+        num_frames = audio_frames.shape[0]
+        # [num_frames, frame_length]
+        audio_signal = torch.tensor(audio_frames, dtype=torch.float32, device=self.device)
+        # [1]
+        audio_signal_len = torch.tensor(num_frames * [self.frame_length], dtype=torch.int32, device=self.device)
+
+        # VAD outputs 2 values for each audio frame with logits indicating the likelihood that
+        # each frame is non-speech or speech, respectively.
+        # [num_frames, 2]
+        log_probs = self.vad_model(input_signal=audio_signal, input_signal_length=audio_signal_len)
+        probs = torch.softmax(log_probs, dim=-1)
+        probs = probs.cpu().detach().numpy()
+        # [num_frames]
+        speech_probs = probs[:, 1]
+        speech_frames = speech_probs >= self.vad_threshold
+
+        return speech_frames
+
+    def _scale_sample_indices(self, start_sample_i: int, end_sample_i: int, sample_rate: int) -> Tuple[int, int]:
+        sample_rate_ratio = sample_rate / self.vad_sample_rate
+        start_sample_i = sample_rate_ratio * start_sample_i
+        end_sample_i = sample_rate_ratio * end_sample_i
+        return start_sample_i, end_sample_i
+
+    def trim_audio(self, audio: np.array, sample_rate: int, audio_id: str = "") -> Tuple[np.array, int, int]:
+        if sample_rate == self.vad_sample_rate:
+            vad_audio = audio
+        else:
+            # Downsample audio to match sample rate of VAD model
+            vad_audio = librosa.resample(audio, orig_sr=sample_rate, target_sr=self.vad_sample_rate)
+
+        if self.volume_norm:
+            # Normalize volume so we have a fixed scale relative to the reference amplitude
+            vad_audio = normalize_volume(audio=vad_audio, volume_level=1.0)
+
+        speech_frames = self._detect_speech(audio=vad_audio)
+
+        start_i, end_i = get_start_and_end_of_speech(
+            is_speech=speech_frames,
+            frame_threshold=self.frame_threshold,
+            frame_step=self.frame_step,
+            audio_id=audio_id,
+        )
+
+        if sample_rate != self.vad_sample_rate:
+            # Convert sample indices back to input sample rate
+            start_i, end_i = self._scale_sample_indices(start_i, end_i, sample_rate)
+
+        start_i, end_i = pad_sample_indices(
+            start_sample_i=start_i,
+            end_sample_i=end_i,
+            max_sample=audio.shape[0],
+            sample_rate=sample_rate,
+            pad_seconds=self.pad_seconds,
+        )
+
+        trimmed_audio = audio[start_i:end_i]
```

Comment:
The +1 is implicitly inside the "get_start_and_end_of_speech" logic (as can be seen in the unit tests for the function). Its a bit hard to follow, but otherwise the +1 needs to be done at the frame level and before the padding rather than in this line at the sample level. I will see if I can make it clearer.